sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,654 @@
+        cross-validation strategies that can be used here.
+
+    stack_method : {'auto', 'predict_proba', 'decision_function', 'predict'}, \
+            default='auto'
+        Methods called for each base estimator. It can be:
+
+        * if 'auto', it will try to invoke, for each estimator,
+          `'predict_proba'`, `'decision_function'` or `'predict'` in that
+          order.
+        * otherwise, one of `'predict_proba'`, `'decision_function'` or
+         `'predict'`. If the method is not implemented by the estimator, it
+         will raise an error.
+
+    n_jobs : int, default=None
+        The number of jobs to run in parallel all `estimators` `fit`.
+        `None` means 1 unless in a `joblib.parallel_backend` context. -1 means
+        using all processors. See Glossary for more details.
+
+    Attributes
+    ----------
+    estimators_ : list of estimators
+        The elements of the estimators parameter, having been fitted on the
+        training data. If an estimator has been set to `'drop'` or None, it
+        will not appear in `estimators_`.
+
+    named_estimators_ : Bunch
+        Attribute to access any fitted sub-estimators by name.
+
+    final_estimator_ : estimator
+        The classifier which predicts given the output of `estimators_`.
+
+    stack_method_ : list of str
+        The method used by each base estimator.
+
+    References
+    ----------
+    .. [1] Wolpert, David H. "Stacked generalization." Neural networks 5.2
+       (1992): 241-259.
+
+    Examples
+    --------
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from sklearn.svm import LinearSVC
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.ensemble import StackingClassifier
+    >>> X, y = load_iris(return_X_y=True)
+    >>> estimators = [
+    ...     ('lr', LogisticRegression(solver='lbfgs', multi_class='auto',
+    ...                               tol=1e-1)),
+    ...     ('svr', LinearSVC(tol=1e-1, random_state=42))
+    ... ]
+    >>> clf = StackingClassifier(
+    ...     estimators=estimators,
+    ...     final_estimator=RandomForestClassifier(n_estimators=10,
+    ...                                            random_state=42),
+    ...     cv=5,
+    ... )
+    >>> from sklearn.model_selection import train_test_split
+    >>> X_train, X_test, y_train, y_test = train_test_split(
+    ... X, y, stratify=y, random_state=42
+    ... )
+    >>> clf.fit(X_train, y_train).score(X_test, y_test)
+    0.9...
+
+    """
+    def __init__(self, estimators, final_estimator=None, cv=None,
+                 stack_method='auto', n_jobs=None, verbose=0):
+        super().__init__(
+            estimators=estimators,
+            final_estimator=final_estimator,
+            cv=cv,
+            stack_method=stack_method,
+            n_jobs=n_jobs,
+            verbose=verbose
+        )
+
+    def _validate_meta_estimator(self):
```

Comment:
what is the meta_estimator refering to? final estimator or self?

Reply:
Shouldn't the estimators also be validated?

Reply:
> what is the meta_estimator refering to? final estimator or self?
 
final_estimator

> Shouldn't the estimators also be validated?

We could. After implementing it, I was wondering if there is a sane case where you want to use a classifier in a stackingregressor or a classifier in a stackingclassifier?

Reply:
I think it's fine to forbid it for now?

Reply:
Yep, we can see if there is need later on.